File: src/audio_steg_desktop/src/AudioSteg.Core/Audio/StegoFileNaming.cs
namespace AudioSteg.Core.Audio;

/// <summary>Default stego WAV filename: stego_YYYY_MM_DD_HHMM_{msg_len}.wav</summary>
public static class StegoFileNaming
{
    public static string Build(int msgBitLength, DateTime? timestamp = null)
    {
        var t = timestamp ?? DateTime.Now;
        return $"stego_{t:yyyy_MM_dd_HHmm}_{msgBitLength}.wav";
    }
}

File: src/audio_stegano_app/lib/core/audio/stego_file_naming.dart
/// Default stego WAV filename: stego_YYYY_MM_DD_HHMM_{msg_len}.wav
String stegoWavFileName(int msgBitLength, [DateTime? timestamp]) {
  final t = timestamp ?? DateTime.now();
  final y = t.year;
  final mo = t.month.toString().padLeft(2, '0');
  final d = t.day.toString().padLeft(2, '0');
  final h = t.hour.toString().padLeft(2, '0');
  final m = t.minute.toString().padLeft(2, '0');
  return 'stego_${y}_${mo}_${d}_$h${m}_$msgBitLength.wav';
}
